src/blockchain/evm/EvmContracts.ts

```diff
@@ -0,0 +1,78 @@
+import type { JsonFragment } from '@ethersproject/abi';
+import { Contract } from 'ethers';
+import { UnsContractName } from 'uns';
+import UnsNetworkConfig from 'uns/uns-config.json';
+
+import { StaticJsonRpcProvider } from './EvmProvider';
+
+type Abi = ReadonlyArray<JsonFragment>;
+
+const getAbi = async <T extends string>(name: T): Promise<Abi> => {
+  try {
+    const lib = await import(`uns/artifacts/abi/${name}.json`);
+    return lib.default;
+  } catch (error) {
+    return [];
+  }
+};
+
+const AbiMapGenerator = <T extends string>(contracts: T[]) => {
+  return Promise.all(
+    Object.values(contracts).map((v) =>
+      getAbi(v).then((abi) => ({ [v]: abi })),
+    ),
+  ).then((a) => a.reduce((v, c) => ({ ...c, ...v }), {} as Record<T, Abi>));
+};
+
+const cache = {} as { [key: string]: Contract };
+
+export type CryptoConfig = Record<
+  string,
+  {
+    address: string;
+    legacyAddresses: ReadonlyArray<string>;
+    getContract: () => Contract;
+  }
+>;
+
+export const getEvmConfig = (
+  networkId: string,
+  abiMap: Record<string, Abi>,
+  provider: StaticJsonRpcProvider,
+  networks: { [key: string]: { contracts: any } },
+): CryptoConfig => {
+  const cryptoConfig = {} as CryptoConfig;
+  if (Object.keys(networks).includes(networkId)) {
+    const { contracts } = networks[networkId];
+
+    Object.keys(contracts).forEach((key) => {
```

Comment:
we should use for in loop for easier to read
and use `contractName` or some better name to say what it is
```suggestion
    for (const contractName in contracts) {
     ...
    }
```